google/cloud/storage/client.cc

```diff
@@ -481,6 +481,10 @@ std::string CreateRandomPrefixName(std::string const& prefix) {
                                                   "abcdefghijklmnopqrstuvwxyz");
 }
 
+std::string Client::ExternalUrl() {
+  return connection_->client_options().endpoint() + "/";
```

Comment:
`client_options()` is deprecated, consider:

```suggestion
  return connection_->options().get<RestEndpoint>();
```

and then add the "/" at the call site.

Reply:
If you omit the `+ "/"` you may be able to return `std::string const&` and save some allocations.